### ISSUE TITLE
Yammer Provider

### ISSUE THREAD

**Issue by rjperes** [TerribleDev/OwinOAuthProviders#70]
Hi,
I am unable to login using the Yammer provider. I have an application created (Web), and an id and secret; when I use it, it redirects to the Yammer dialog page, and after I accept it, it goes back to the login page, without any error message. By inspecting the HTTP results, I see that I am getting an access denied response.
I thought of two possible explanations:

1) My return URL wasn't right; so, I tried several possibilities, either with and without a trailing /signin-Yammer, but no luck; this URL is public and fully qualified, BTW;
2) Yammer forces me to add my app to either the local or the global directory, which I haven't done, because at this stage this is just for learning purposes.

Any thoughts?

Thanks in advance, and keep up the great work!

RP

**Comment by AlexArchive** [comment]
I will inspect this soon.

I think it was @ravind who wrote this provider. @ravind, might you know what the problem here is? Thank you.

**Comment by ravind** [comment]
1) Have you tried in more than one browser? I've seen people have browser specific issues before.
2) Have you configured it to limit access to certain Yammer networks? If so please make sure you're logged into the correct network on Yammer before trying to login to your site.

**Comment by rjperes** [comment]
Hi, @ravind!
Thanks. I tested it with IE, Firefox and Chrome, same problem.
What is weird is that I receive an email notification from Yammer saying that I logged in "from a new application or browser"!
What is this thing about restricting networks? I surely haven't done anything like that. Can you login successfully? What return url have you registered with Yammer?

Thanks!

**Comment by ravind** [comment]
Yes, I've been able to login successfully and am using it on more than one site. The Redirect URI on Yammer just points to my root path (E.g. https://subdomain.azurewebsites.net/).

The provider has an "AcceptedNetworks" configuration option which allows you to limit access based on the user's Yammer network. E.g. If you only wanted to allow people from your company to access the app, you could set it up as follows in your Startup.Auth.cs:

```
        app.UseYammerAuthentication(new YammerAuthenticationOptions
        {
            ClientId = ConfigurationManager.AppSettings["YammerClientId"],
            ClientSecret = ConfigurationManager.AppSettings["YammerClientSecret"],
            AcceptedNetworks = new string[] { "CompanyNetworkName" }
        });
```

**Comment by rjperes** [comment]
Weird... I use a public domain as well, and a port... no accepted networks.
Is your application deployed locally or globally?

**Comment by ravind** [comment]
It works for me both locally and when deployed to a public URL, though you need to update the Redirect URI on Yammer accordingly. I ended up registering two applications in Yammer, one for the production URL and a "Dev" application using localhost.

**Comment by rjperes** [comment]
That's not what I asked... Yammer applications can be deployed to a local or a global directory. This is done on their site, and it requires filling a number of fields. Are yours deployed this way?

**Comment by ravind** [comment]
Sorry, I misunderstood. I have not deployed my apps to either the local or global Yammer directory.

**Comment by rjperes** [comment]
Hmm, in that case, I'm officially clueless! :-)
It's weird because I don't actually see an error message, I'm just returned to the login page unauthenticated. It's just by inspecting the responses that I see the access denied error. And Yammer, AFAIK, only answers to technical questions on StackOverflow, and I asked there some says ago, and got no reply till now.
